gdx/src/com/badlogic/gdx/graphics/g3d/shaders/BaseShader.java

```diff
@@ -113,6 +113,15 @@ public boolean validate (final BaseShader shader, final int inputID, final Rende
 	public Camera camera;
 	private Mesh currentMesh;
 
+	/** ClassX: thread-safety support */
+	public final Matrix4 tempMatrix4 = new Matrix4();
+
+	/** ClassX: thread-safety support */
+	public final Matrix3 tempMatrix3 = new Matrix3();
+
+	/** ClassX: thread-safety support */
+	public final Vector3 tempVector3 = new Vector3();
```

Comment:
`protected` also here?

Reply:
@dar-dev

Reply:
this _Vector3_ is referenced outside of BaseShader (i.e. ParticleShader).. so unfortunately I cannot change the visibility